plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/guru/VMwareGuru.java

```diff
@@ -1104,13 +1124,22 @@ public List<Command> finalizeMigrate(VirtualMachine vm, Map<Volume, StoragePool>
             if (pool.getClusterId() != null) {
                 poolClusterId = pool.getClusterId();
             }
+            if (volume.getVolumeType().equals(Volume.Type.ROOT) && pool.isLocal()) {
+                isLocalStorageMigration = true;
+            }
             volumeToFilerTo.add(new Pair<VolumeTO, StorageFilerTO>(volumeTo, filerTo));
         }
         final Long destClusterId = poolClusterId;
         final Long srcClusterId = vmManager.findClusterAndHostIdForVm(vm.getId()).first();
         final boolean isInterClusterMigration = isInterClusterMigration(destClusterId, srcClusterId);
+        String targetHostGuid = null;
+        if (isLocalStorageMigration) {
```

Comment:
if above suggestion is addressed, change this here. thanks.

```suggestion
        if (targetVmPool != null && targetVmPool.isisLocal()) {
            targetHostGuid = getHostGuidForLocalStorage(targetVmPool.getId());
```